src/language-js/print/call-arguments.js

```diff
@@ -289,4 +284,15 @@ function shouldGroupFirstArg(args) {
   );
 }
 
+function isReactHookCallWithDepsArray(args, isArrowBodyBlock) {
+  return (
+    args.length === 2 &&
+    args[0].type === "ArrowFunctionExpression" &&
+    getFunctionParameters(args[0]).length === 0 &&
```

Comment:
Maybe we can allow some simple parameters, so we can format this case the same way

**Prettier pr-10238**
[Playground link](https://deploy-preview-10238--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuc0DOMAEBtTAzASwCcNMBdTAXkwEMiiaBPAOiLgBMBXMOACgB0omTLwCUVAHw4aYMJwC2nADY0YEIgBpMcJXHkIYWmXMUq1m7bv2wyGwcOx5lSgHI19ZQaIDcgwZChSXEISLApqOgYWNi4eASERYwVlVXUtHT0DLXFKKWwk01SLDOsYW3scJyVXdzhPKB8QDRAIAAcYAnRkUEiIAHcABToENGQQGiU+plHmgCMGMABrOBgAZVaZAigAc2QYIk44Zr1ZjnYOABkaHc4abbgAMXV5VQ6dsZpONSaQAAsYeRKADqvwI8DQGx4qxGYIIADcwYwxmA0DMQFs0HAiDABgxti9kHgJpjmgArNAADwAQgtlmtahctnBCcSjiByRTVlttroAIqcCDwFlKEkgDYkLFjWY0U5KH6tIhbGBAgjsGC-ZAADgADM0FRBMUCGK0xgq4JiiHDmc0AI4C+C4tqjFA0NAAWigcA4HB+bDtxDguLuBKQRJFbMx8gIewOEe5fPtzNDrOaMBlKrVGqQACZUwwCEpuQBhCDyEMgc0AVh+nExABUZc6w6K4YcAJJQc6wVZgRXtACCndWMEYumFmIAvhOgA)
```sh
--parser babel
```

**Input:**
```jsx
const [ first ] = array.reduce(
  () => [accumulator, element, accumulator, element],
  [fullName]
);

const [ first ] = array.reduce(
  (accumulator, element, ) => [accumulator, element],
  [fullName]
);
```

**Output:**
```jsx
const [first] = array.reduce(
  () => [accumulator, element, accumulator, element],
  [fullName]
);

const [first] = array.reduce((accumulator, element) => [accumulator, element], [
  fullName,
]);

```

Reply:
Let's just prevent hugging of the last argument if:
```js
args.length === 2 &&
args[0].type === "ArrowFunctionExpression" &&
args[1].type === "ArrayExpression"
```
It's difficult for me to imagine a case when last arg hugging would look good in such a call.